drivers/can/Kconfig.mcan

```diff
@@ -20,4 +21,16 @@ config CAN_DELAY_COMP
 	help
 	  Enable the automatic transceiver delay compensation.
 
+config CAN_MCAN_MSG_RAM_ADDR_ABSOLUTE
+	bool
+	# Hidden option for individual driver implementation.
+	# Addresses in Message RAM are absolute.
+	default n
+
+config CAN_MCAN_MSG_RAM_ADDR_RELATIVE
+	bool
+	# Hidden option for individual driver implementation.
+	# Addresses in Message RAM are relative to the Message RAM base address.
+	default n
```

Comment:
That's a good point. I've taken a shot at this, adding a `msg-ram-absolute-addressing` property to bosch,m-can-base.yaml, since I believe this rule applies to the whole device, not just as particular instance.

Reply:
Actually, I'd like to suggest a third approach instead. Since we are already using vendor-specific compatible strings in the dts, this information was already given (e.g. the NXP LPC MCAN implementation always uses relative addresses) so there is no need for an extra dts property for this. Rather, this should be filled into `struct can_mcan_config` by the vendor-specific driver frontend likely as some sort of flags field, since we anticipate more variance points to be identified as more vendor frontend drivers are added.

There are other variance points already identified (extra vendor-specific registers, extra initialisation code, ...) and the fact that the current driver approach doesn't support multiple instances in a good way (the layout of the structs are based on the dts properties of the first instance) and so an so forth.

Seeing we already have three implementations using the current architecture (the STM32FD one in tree, the STM32H7 in this PR and the NXP LPC in #35832) I suggest (as already discussed on Discord) we get all of these in-tree with more or less the architecture of the current mcan driver - and _then_ start identifying variance points, refactor, clean up etc. With just two frontends handled in this PR (the existing STM32FD and the STM32H7) we are likely to make some wrong decisions on how to refactor the driver to handle the variance as we don't have a non-ST vendor implementation represented.

Reply:
@henrikbrixandersen I think that suggestion makes even more sense, as you say it's not like you have a choice to use one addressing mode or the other with a particular part.

For the particular code in question, we don't even have an existing implementation using it—the STM32FD doesn't use that path. The way it is now, with absolute addresses, was meant to target an implementation that wasn't completed.

So, I take it that you're thinking we should use the vendor-specific defines temporarily to get these two/three implementations working, and then refine things in a future step?

So we might say:

```
	uint32_t mrba = 0;
#ifdef CONFIG_CAN_STM32H7
	mrba = (uint32_t)msg_ram;
#endif
#ifdef CONFIG_ CAN_MCUX_MCAN
	mrba = (uint32_t)msg_ram & CAN_MCAN_MRBA_BA_MSK;
#endif
```

Reply:
We could also split out the part that sets the msg_ram address/msg_ram offset in a weak function that implements the behavior for the "original" m_can. The frontend that needs special treatment can just override it.
The idea from @henrikbrixandersen to put it into the config has also a valid point. But there i would rather put some functionpointers in to not bloat the code.